server/docs/design/block-verification.md

```diff
@@ -0,0 +1,193 @@
+1. The `BlockStatusManager` initiates any necessary recovery or follow-up processes depending on the verification result.
+
+Sequence Diagram:
+```mermaid
+sequenceDiagram
+    participant U as UnverifiedRingBuffer
+    participant V as VerificationHandler
+    participant F as BlockHashingSessionFactory
+    participant S as BlockHashingSession
+    participant U as UnverifiedRingBuffer
+    participant V as VerificationHandler
+    participant F as BlockHashingSessionFactory
+    participant S as BlockHashingSession
+    participant SV as SignatureVerifier
+    participant BSM as BlockStatusManager
+        
+    U->>V: (1) onBlockItemsReceived(List<BlockItem>)        
+    
+    alt (2) Detects block_header
+    
+    V->>F: createSession(initialBlockItems, executorService, signatureVerifier)
+    Note over S: New instance of BlockHashingSession created
+    F-->>V: returns BlockHashingSession (S)
+    V-->>U: Returns without blocking
+    S->>S: *Starts hash computation asynchronously
+    
+    else (3) Append more Block Items
+    loop
+    V->>S: addBlockItems(items)
+    V-->>U: return without blocking
+    S->>S: *Continues hash computation asynchronously
+    end
+
+    else (4) Append BlockItems with block_proof
+   
+    V->>S: addBlockItems(items with block_proof)
+    V-->>U: return without blocking    
+    S->>S: async completeHashing()   
+    S->>SV: (5) verifySignature(signature, computedHash, blockNumber)    
+
+    Note over SV,BSM: (6) Compare computed hash and signature
+    alt (7) Verification Fails
+      SV->>BSM: updateBlockStatus(blockNumber, ERROR, SIGNATURE_INVALID      
+    else (8) Verification Succeeds
+      SV->>BSM: updateBlockStatus(blockNumber, VERIFIED, NONE)      
+    end
+    Note over BSM: (9) Follow-up to downstream services
+    end
+
+```
+
+## Interfaces
+
+### VerificationHandler
+```java
+public interface VerificationHandler {
+  void onBlockItemsReceived(List<BlockItem> blockItems);  
+}
+```
+
+### BlockHashingSessionFactory
+```java
+import java.util.concurrent.ExecutorService;
+
+public interface BlockHashingSessionFactory {
+  BlockHashingSession createSession(List<BlockItem> initialBlockItems, ExecutorService executorService, SignatureVerifier signatureVerifier);
+}
+```
+
+### BlockHashingSession
+```java
+public interface BlockHashingSession {
+  void addBlockItem(BlockItem item);
+  CompletableFuture<Void> completeHashing();
+}
+```
+
+### SignatureVerifier
+```java
+public interface SignatureVerifier {
+  void verifySignature(byte[] signature, byte[] computedHash, long blockNumber);  
+}
+```
+
+### BlockStatusManager
+```java
+
+public interface BlockStatusManager {
+    void updateBlockStatus(long blockNumber, BlockVerificationStatus status, VerificationError error);
+}
+```
+
+## Enums
+```java
+public enum BlockVerificationStatus {
+  VERIFIED,
+  ERROR
+}
+
+public enum VerificationError {
+  NONE,
+  SIGNATURE_INVALID,
+  SYSTEM_ERROR
+}
+```
+
+## Configuration
```

Comment:
I agree that we should not duplicate documentation, as in the case of keeping configuration here and in the Config document.

I added this section since we need to highlight that we will need a `VerificationConfig` Object and that we need to include at least a way to enable or disable the verification handler, agreed that the final implementation will change how to do it and even the name and possible values for the configurations. so we want to just note that we will take into consideration some but not limited to, configuration properties.

✅ Done!